specs/language/introduction.tex

```diff
@@ -77,46 +118,67 @@
 simultaneously.
 
-\Sub{\gls{wave}}{Intro.Model.Wave}
-
-\p A grouping of \gls{lane}s for execution is called a \gls{wave}. \gls{wave}
-sizes vary by hardware architecture. Some hardware implementations support
-multiple wave sizes. Generally wave sizes are powers of two, but there is no
-requirement that be the case. \acrshort{hlsl} is explicitly designed to run on
-hardware with arbitrary \gls{wave} sizes.
-
-\Sub{\gls{quad}}{Intro.Model.Quad}
-
-\p A \gls{quad} is a subdivision of four \gls{lane}s in a \gls{wave} which are
+\Par{\gls{wave}}{Intro.Model.Terms.Wave} A grouping of \gls{lane}s for execution
+is called a \gls{wave}. \gls{wave} sizes vary by hardware architecture. Some
+hardware implementations support multiple wave sizes. Generally wave sizes are
+powers of two, but there is no requirement that be the case. \acrshort{hlsl} is
+explicitly designed to run on hardware with arbitrary \gls{wave} sizes. Hardware
+architectures may implement \gls{wave}s as \acrfull{simt} where each thread
+executes instructions in lock-step. This is not a requirement of the model. Some
+constructs in \acrshort{hlsl} require synchronized execution. Such constructs
+will explicitly specify that requirement.
+
+\Par{\gls{quad}}{Intro.Model.Terms.Quad}
+A \gls{quad} is a subdivision of four \gls{lane}s in a \gls{wave} which are
 computing adjacent values. In pixel shaders a \gls{quad} may represent four
 adjacent pixels and \gls{quad} operations allow passing data between adjacent
 lanes. In compute shaders quads may be one or two dimensional depending on the
 workload dimensionality described in the \texttt{numthreads} attribute on the
 entry function. (FIXME: Add reference to attribute)
 
-\Sub{\gls{threadgroup}}{Intro.Model.Group}
+\Par{\gls{threadgroup}}{Intro.Model.Terms.Group}
+A grouping of \gls{lane}s executing the same shader to produce a combined
+result is called a \gls{threadgroup}. \gls{threadgroup}s are independent of
+\acrshort{simd} hardware specifications. The dimensions of a \gls{threadgroup}
+are defined in three dimensions. The maximum dimensions of a \gls{threadgroup}
+are defined by the runtime.
 
-\p A grouping of \gls{wave}s executing the same shader to produce a combined
-result is called a \gls{threadgroup}. \gls{threadgroup}s are executed on
-separate \acrshort{simd} hardware and are not instruction locked with other
-\gls{threadgroup}s.
-
-\Sub{\gls{dispatch}}{Intro.Model.Dispatch}
-
-\p A grouping of \gls{threadgroup}s which represents the full execution of a
+\Par{\gls{dispatch}}{Intro.Model.Terms.Dispatch}
+A grouping of \gls{threadgroup}s which represents the full execution of a
 \acrshort{hlsl} program and results in a completed result for all input data
 elements.
 
+\Sub{\acrshort{spmd} Execution Model}{Intro.Model.Exec}
+
+\p A runtime implementation will provide an implementation-defined mechanism for
+defining a \gls{dispatch}. In an implementation-defined way a runtime will
```

Comment:
I might rephrase this a bit, although I'm not sure what the best phrasing is. I think I would naturally write something like:

> Implementations are permitted to vary with respect to hardware resource management and execution scheduling provided that the observed behavior conforms to the behavior described in this specification. For example, a runtime implementation may choose to...